glusterd2/xlator/options/options.go

```diff
@@ -250,8 +250,16 @@ func ValidatePercentOrSize(o *Option, val string) error {
 }
 
 // ValidatePriorityOrSize validates either priority or size
+// Example: val := "k1:1024KB, k2:512MB, k3:512GB"
+// It is verified next if 1024KB is a valid size.
 func ValidatePriorityOrSize(o *Option, val string) error {
-	return ErrInvalidArg
+	pairs := strings.Split(val, ",")
+	for _, pair := range pairs {
+		kv := strings.Split(pair, ":")
+		err := ValidateSizeList(o, kv[1])
```

Comment:
Depending on the suffix, you'll need to either validate size or just ensure the value is a number. ValidateSizeList validates that the value is a number, but it also tries to validate that number is a multiple of 512, so it doesn't work.

Just that one addition.

Reply:
@kshlm I have added validation checks for both size and number.
And for the second comment, I don't think our efforts to break ValidateSizeList() into two functions(say validateSizeType) would help, since we'll anyway have to check if the string has a 'B' suffix to call ValidateSizeType() in both the caller functions.
Please review if we wish to implement this^^ Thanks.